mmhuman3d/data/datasets/human_image_dataset.py

```diff
@@ -81,39 +94,71 @@ def load_annotations(self):
-        self.num_data = self.human_data.data_len
+        if self.cache_data_path is None:
+            use_human_data = True
+        elif rank == 0 and not os.path.exists(self.cache_data_path):
+            use_human_data = True
+        else:
+            use_human_data = False
+        if use_human_data:
+            self.human_data = HumanData.fromfile(self.ann_file)
+
+            if self.human_data.check_keypoints_compressed():
+                self.human_data.decompress_keypoints()
+            # change keypoint from 'human_data' to the given convention
+            if 'keypoints3d' in self.human_data:
+                keypoints3d = self.human_data['keypoints3d']
+                assert 'keypoints3d_mask' in self.human_data
+                keypoints3d_mask = self.human_data['keypoints3d_mask']
+                keypoints3d, keypoints3d_mask = \
+                    convert_kps(
+                        keypoints3d,
+                        src='human_data',
+                        dst=self.convention,
+                        mask=keypoints3d_mask)
+                self.human_data.__setitem__('keypoints3d', keypoints3d)
+                self.human_data.__setitem__('keypoints3d_convention',
+                                            self.convention)
+                self.human_data.__setitem__('keypoints3d_mask',
+                                            keypoints3d_mask)
+            if 'keypoints2d' in self.human_data:
+                keypoints2d = self.human_data['keypoints2d']
+                assert 'keypoints2d_mask' in self.human_data
+                keypoints2d_mask = self.human_data['keypoints2d_mask']
+                keypoints2d, keypoints2d_mask = \
+                    convert_kps(
+                        keypoints2d,
+                        src='human_data',
+                        dst=self.convention,
+                        mask=keypoints2d_mask)
+                self.human_data.__setitem__('keypoints2d', keypoints2d)
+                self.human_data.__setitem__('keypoints2d_convention',
+                                            self.convention)
+                self.human_data.__setitem__('keypoints2d_mask',
+                                            keypoints2d_mask)
+            self.human_data.compress_keypoints_by_mask()
+
+        if self.cache_data_path is not None:
+            if rank == 0 and not os.path.exists(self.cache_data_path):
+                writer_kwargs, sliced_data = self.human_data.get_sliced_cache()
+                writer = HumanDataCacheWriter(**writer_kwargs)
+                writer.update_sliced_dict(sliced_data)
+                writer.dump(self.cache_data_path)
+            dist.barrier()
+            self.reader = HumanDataCacheReader(npz_path=self.cache_data_path)
+            self.num_data = self.reader.data_len
+            self.human_data = None
+        else:
+            self.reader = None
+            self.num_data = self.human_data.data_len
```

Comment:
We may add unit test to cover the cache reading part in [test_human_image_dataset.py](https://github.com/open-mmlab/mmhuman3d/blob/main/tests/test_datasets/test_human_image_dataset.py)

Reply:
unit test has been added in the latest commit